dev/AppNotifications/WICUtility.cpp

```diff
@@ -0,0 +1,165 @@
+﻿
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "WICUtility.h"
+#include <cwctype>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/base.h>
+#include <externs.h>
+
+#include <ShObjIdl_core.h>
+#include <wincodec.h>
+#include <Ocidl.h>
+#include <windows.h>
+
+typedef enum
+{
+    BMPV_1,
+    BMPV_5,
+} BITMAP_VERSION;
+
+winrt::com_ptr<IWICBitmapSource> ConvertWICBitmapPixelFormat(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    WICPixelFormatGUID guidPixelFormatSource,
+    WICBitmapDitherType bitmapDitherType)
+{
+    winrt::com_ptr<IWICFormatConverter> wicFormatConverter;
+    THROW_IF_FAILED(wicImagingFactory->CreateFormatConverter(wicFormatConverter.put()));
+
+    THROW_IF_FAILED(wicFormatConverter->Initialize(wicBitmapSource.get(), guidPixelFormatSource, bitmapDitherType, nullptr, 0.f, WICBitmapPaletteTypeCustom));
+
+    // Store the converted bitmap as ppToRenderBitmapSource
+    winrt::com_ptr<IWICBitmapSource> wicBitmapSourceConverted;
+    THROW_IF_FAILED(wicFormatConverter->QueryInterface(_uuidof(IWICBitmapSource), wicBitmapSourceConverted.put_void()));
+
+    return wicBitmapSourceConverted;
+}
+
+void AddFrameToWICBitmap(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapEncoder> const& wicEncoder,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    BITMAP_VERSION bmpv)
+{
+    winrt::com_ptr<IWICBitmapFrameEncode> wicFrameEncoder;
+    winrt::com_ptr<IPropertyBag2> wicEncoderOptions;
+
+    THROW_IF_FAILED(wicEncoder->CreateNewFrame(wicFrameEncoder.put(), wicEncoderOptions.put()));
+
+    GUID containerGuid;
```

Comment:
winrt::guid?

Reply:
Unfortunately, a `winrt::guid` cannot be passed in the place of a GUID*, so will leave it as is.